### DISCUSSION TITLE
Ignore SmartStart inclusion requests when S2 keys are not configured

### DISCUSSION THREAD

**Discussion by kpine** [zwave-js/zwave-js#6644] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

If S2 keys are not configured and a user includes a node with SmartStart, the S2 bootstrapping fails (presumably, but I didn't see the logs). This problem was reported by a user in HA Discord who reported the "infinite" inclusion using SmartStart, and the problem was solved by adding S2 keys, which were previously not set.

According to one of the spec docs [1], on S2 bootstrapping failure:

> The joining node may continue with SmartStart learn mode until successful SmartStart inclusion or re-try at least up to 2 times and give up.

Kind of a weird phrasing, but it sounds like the device can either try forever, or try for a minimum of 2 times and then give up. It seems this device (ZEN04) tries forever.

So I guess it doesn't it doesn't make much sense to attempt an inclusion without S2 keys as SmartStart requires S2. I did not find anything in the spec docs that alludes to this use case.

[1] Section "4.5.4.3.4 Unsuccessful SmartStart inclusion" in Z-Wave and Z-Wave Long Range Network Layer Specification 2.0.

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [ ] The log includes a re-interview of the problematic device _(if applicable)_.

- [ ] The log includes the problematic interaction with the device _(if applicable)_.

- [ ] I provided the node ID of the problematic device _(if applicable)_.

### Upload Logfile

None provided by user.

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment, marked as answer]
Escalated to an issue https://github.com/zwave-js/node-zwave-js/issues/6771
